functions.py

```diff
@@ -47,7 +50,7 @@ def get_all_repos(username):
     all_repos = redis_client.get(f"{username}"+"_all_repos")
     if all_repos is None:
         try:
-            g = Github(base_url="https://api.github.com")
+            g = Github(base_url=URL)
             all_repos = []
             user = g.get_user(username)
```

Comment:
**Suggestion:** Refactor the code to create a function for getting a Github user. This will avoid code duplication.
```suggestion
            # Use the get_user function instead
            user = get_user(username)
```